bot.py

```diff
@@ -137,13 +113,42 @@ async def fetch_message_by_id(self, channel_id, message_id):
             return None
 
 
-def run_bot():
-    intents = discord.Intents.default()
-    intents.message_content = True
-    intents.voice_states = True
-    client = Khaled(intents=intents)
-    client.run(token)
-
+async def main():
+    logger = logging.getLogger('discord')
+    logger.setLevel(logging.INFO)
+
+    handler = logging.handlers.RotatingFileHandler(
+        filename='discord.log',
+        encoding='utf-8',
+        maxBytes=32 * 1024 * 1024,  # 32 MiB
+        backupCount=5,  # Rotate through 5 files
+    )
+    dt_fmt = '%Y-%m-%d %H:%M:%S'
+    formatter = logging.Formatter('[{asctime}] [{levelname:<8}] {name}: {message}', dt_fmt, style='{')
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+    with open("config.json") as f:
+        config = json.load(f)
+        token = config["secrets"]["discordToken"]
+        exts = [
+            'cogs.btc',
+            'cogs.chess_leaderboard',
+            'cogs.chess',
+            'cogs.database',
+            'cogs.div',
+            'cogs.emoji', 
+            'cogs.files', 
+            'cogs.leaderboard', 
+            'cogs.music', 
+            'cogs.purge', 
+            'cogs.restart', 
+            'cogs.status'
+        ]
+        intents = discord.Intents.default()
+        intents.message_content = True
+        intents.voice_states = True
+        async with Khaled("?", intents=intents, initial_extensions=exts) as bot:
```

Comment:
can we not use a prefix?

Reply:
empty string works fine, current value is for testing i'm going to keep the original one, documentation: An empty string as the prefix always matches, enabling prefix-less command invocation. While this may be useful in DMs it should be avoided in servers, as it’s likely to cause performance issues and unintended command invocations. https://discordpy.readthedocs.io/en/stable/ext/commands/api.html?highlight=prefix#discord.ext.commands.Bot.command_prefix